src/librustdoc/html/render/cache.rs

```diff
@@ -226,18 +226,25 @@ fn get_index_type_name(clean_type: &clean::Type, accept_generic: bool) -> Option
             let path = &bounds[0].trait_;
             Some(path.segments.last().unwrap().name)
         }
+        clean::ImplTrait(ref bounds) => {
+            let first_trait = bounds.iter().find_map(|b| match b {
+                clean::GenericBound::TraitBound(poly_trait, _) => Some(&poly_trait.trait_),
```

Comment:
Note that this will always pick the first trait bound, even if it's `?Sized` or `~const Drop`.

Reply:
It should return all trait, not just the first one. I have no idea what `~const Drop` means and can't find it on the internet... If it's like `!Drop`, we shouldn't list it. For `?Sized`, I'm not sure if we should list it or not...

Reply:
I agree, though note that the pre-existing code only returns the first trait of a `dyn Trait1 + Trait2 + ...` type. Making it return all traits would be a bigger change; would you like me to attempt that here?

`~const Drop`: #67792

Reply:
> Making it return all traits would be a bigger change; would you like me to attempt that here?

Yes please!

Reply:
Done!